stistools/defringe/normspflat.py

```diff
@@ -41,58 +52,309 @@ def normspflat(inflat, outflat, do_cal=True, biasfile=None, darkfile=None, pixel
+                ', '.join(keyword_warnings)))
+
+    # Read in the calibrated flat data:
+    data = fits.getdata(outname, ext=1)
+    numrows, numcols = np.shape(data)
 
     # Do a line-by-line cubic spline fit to the fringe flat to remove the lamp function...
 
-    # If short-slit aperture (does not start with "52X"):
-    #    Find the row with max counts in the short-slit fringe flat
-    #    Uses central 60% of columns
-    #    Does some flux-filtering
+    with fits.open(outname) as hdulist:
+
+        aperture = hdulist[0].header['APERTURE']
+        bincols = hdulist[0].header['BINAXIS1']
+        binrows = hdulist[0].header['BINAXIS2']
+        cenwave = hdulist[0].header['CENWAVE']
+
+        # If short-slit aperture (does not start with "52X"):
+        if aperture[0:3] != "52X":
+            flatdata = hdulist[1].data
+            #    Find the row with max counts in the short-slit fringe flat
+            #    Search between the middle 10% of rows and the middle 60% of columns
+            t_row = int(0.45 * numrows)
+            b_row = int(0.55 * numrows)
+            #     Uses central 60% of column
+            l_col = int(0.2 * numcols)
+            r_col = int(0.8 * numcols)
+
+            row_avgs = np.array([np.average(row) for row in flatdata[t_row:b_row, l_col:r_col]])
+            max_row_idx = np.where(row_avgs == np.max(abs(row_avgs)))[0][0] + t_row  # CL does an absolute value here
+            max_row = flatdata[max_row_idx]
+            #    Does some flux-filtering -- I think this is for determining the max rows
+
+
+        # Set rows (startrow, lastrow) to be fit according to aperture name (and possibly OPT_ELEM):
+        # '0.3X0.09', '0.2X0.06', '52X...'
+        if aperture == "0.3X0.09":
+            startrow = max_row_idx - int(4./binrows+0.25)
+            lastrow = max_row_idx + int(4./binrows+0.25)
+        elif aperture == "0.2X0.06":
+            startrow = max_row_idx - int(3. / binrows + 0.25)
+            lastrow = max_row_idx + int(3. / binrows + 0.25)
+        elif aperture[0:3] != "52X":
+            print("ERROR: not able to understand APERTURE keyword")
+            return
+        elif opt_elem == "G750M":
+            startrow = int(92./binrows) + 1
+            lastrow = startrow + int(1024./binrows) - 1
+        else:
+            startrow = 1
+            lastrow = numrows
+
+        # Details of spline fit determined according to OPT_ELEM + CENWAVE.
+        # G750M (various CENWAVEs), G750L (i.e. CENWAVE == 7751; various binning), other (never used?)
+        if opt_elem == "G750M":
+            fitted = []  # This probably needs to be padded with ones to match the input shape
+
+            if cenwave < 9800:
+                for idx, row in enumerate(data):
+                    if (idx >= startrow) and (idx <= lastrow):
+                        l_row_data = row[0:86]
+                        r_row_data = row[1109:]
+                        fit_data = row[86:1109]
```

Comment:
In Python slices such as this, you probably should increase the second number by one, in comparison with the cl script.